spring-integration-redis/src/test/java/org/springframework/integration/redis/util/RedisLockRegistryTests.java

```diff
@@ -384,6 +385,18 @@ public void testExpireNoLockInStore() throws Exception {
 
 	@Test
 	@RedisAvailable
+	public void testExpireDuringSecondObtain() throws Exception {
```

Comment:
Thank you for feedback, @rpresnakov !

I've just added one more assert there. See the current version of this test.

Regarding your use-case...

Sounds correct and as expected behavior in the same JVM with normal `Lock`.

Here we really can't know if the Redis value has been expired until we ask for that, like in case of `unlock()`.

That will work as you expect if you use different `RedisLockRegistry` instances.
Otherwise everything should look like you use `DefaultLockRegistry`.

We really can't unlock `java.util.concurrent.locks.ReentrantLock` from some other Thread, unless owner...